packages/teams-js/test/private/messageChannels.spec.ts

```diff
@@ -40,27 +41,100 @@ describe('messageChannels', () => {
       expect(() => messageChannels.isSupported()).toThrowError(new Error(errorLibraryNotInitialized));
     });
 
+    it('isSupported for telemetry should throw if called before initialization', () => {
+      utils.uninitializeRuntimeConfig();
+      expect(() => messageChannels.telemetry.isSupported()).toThrowError(new Error(errorLibraryNotInitialized));
+    });
+
+    it('isSupported for data layer should throw if called before initialization', () => {
+      utils.uninitializeRuntimeConfig();
+      expect(() => messageChannels.dataLayer.isSupported()).toThrowError(new Error(errorLibraryNotInitialized));
+    });
+
     it('getTelemetryPort should throw if called before initialization', async () => {
       expect.assertions(1);
       utils.uninitializeRuntimeConfig();
-      await expect(messageChannels.getTelemetryPort()).rejects.toThrowError(new Error(errorLibraryNotInitialized));
+      await expect(messageChannels.telemetry.getTelemetryPort()).rejects.toThrowError(
+        new Error(errorLibraryNotInitialized),
+      );
+    });
+
+    it('getDataLayerPort should throw if called before initialization', async () => {
+      expect.assertions(1);
+      utils.uninitializeRuntimeConfig();
+      await expect(messageChannels.dataLayer.getDataLayerPort()).rejects.toThrowError(
+        new Error(errorLibraryNotInitialized),
+      );
     });
   });
 
   Object.values(FrameContexts).forEach((context) => {
-    describe('Testing isSupported', () => {
-      it('should return true if the capability is supported', async () => {
+    describe('Testing messageChannels isSupported', () => {
```

Comment:
whats the value in running through each frameContext? the code under test doesnt use this

Reply:
This was a requirement from the teams-js team on my previous PR (I had the same question). Some APIs change based on frame context, so it was recommended to make sure we are testing this just in case some one starts introducing changes in this area.

Reply:
Yes, kerryn is correct. Really, sumathi is too and for this particular code right now there's no difference. Testing for each framecontext will make sure that if someone goes and removes one in the future it will break a test (which is good because removing a frame context would be a breaking change and we don't want to do that accidentally).

Ultimately, I would like to get rid of the framecontext concept entirely and depend entirely on isSupported, allowing the hosts to control what is and isn't supported based on their knowledge of where the app is running, but there are one or two things we'd need to change before we get there